### ISSUE TITLE
Switch to patch -p1 format by default

### ISSUE THREAD

**Issue by Gottox** [voidlinux/void-packages#10875]
The world uses `-p1` format for patches. Unfortunately, xbps-src uses the `-p0` format which is a historic remnant from it's NetBSD roots.

I'm proposing to change it to `-p1` format by default to make importing patches from other sources simpler.

**Comment by maxice8** [comment]
Will it break patches that go directly to dest/ ?

**Comment by Gottox** [comment]
All templates that do not define `patch_args` will break, but I can work out a regex to fix the patches afterwards

**Comment by pullmoll** [comment]
For me either format is ok. I don't usually use git to create a diff but diff itself on a local copy. OTOH it's never to late to change an attitude :)

**Comment by maxice8** [comment]
On Tue, Jan 16, 2018 at 09:21:28AM +0000, Enno Boland wrote:
> All templates that do not define `patch_args` will break, but I can work out a regex to fix the patches afterwards
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/voidlinux/void-packages/issues/10875#issuecomment-357900023

oh that's sounds nice

**Comment by ebfe** [comment]
-p0 patches are more convenient to create

**Comment by pullmoll** [comment]
Perhaps we can write an `xtools` helper like `xdiffdepth [0|1] <template>` which would modify patches of a package by prepending or removing e.g. the {`${pkgname}-${version}/`,.orig} or `a/` `b/`.

**Comment by ebfe** [comment]
If the main issue is that git diffs have a prefix by default there's '--no-prefix'

**Comment by Gottox** [comment]
What's with the import of patches from other vendors?

**Comment by ebfe** [comment]
> What's with the import of patches from other vendors?

Then you set patch_args, create patch.args or adjust them.

**Comment by Gottox** [comment]
That's not the question I asked.

**Comment by Gottox** [comment]
Let me rephrase: Is importing patches from other sources common enough to streamline this workflow?

**Comment by chneukirchen** [comment]
+1

almost all foreign patches are -p1, and it's just as easy to create them with unpatch.
If you use normal diff, my diffed can now add them easily. :)
